lib/string.gi

```diff
@@ -642,23 +642,23 @@ InstallGlobalFunction("RemoveCharacters", REMOVE_CHARACTERS);
 ##
 #F  EvalString( <expr> ) . . . . . . . . . . . . evaluate a string expression
 ##
-_EVALSTRINGTMP := 0;
-InstallGlobalFunction("EvalString", function( s )
-  local a, f, res;
-  a := "_EVALSTRINGTMP:=";
-  Append(a, s);
-  Add(a, ';');
-  Unbind(_EVALSTRINGTMP);
-  f := InputTextString(a);
-  Read(f);
-  if not IsBound(_EVALSTRINGTMP) then
-    Error("Could not evaluate string.\n");
-  fi;
-  res := _EVALSTRINGTMP;
-  Unbind(_EVALSTRINGTMP);
-  return res;
+InstallGlobalFunction("EvalString",
+function(string)
+    local instream, outstream, instring, outstring, res;
+    instring := Concatenation(string, ";");
+    outstring := "";
+    instream := InputTextString(string);
+    outstream := OutputTextString(outstring, true);
+
+    SET_OUTPUT(outstream, true);
+    res := READ_ALL_COMMANDS(instream, false);
```

Comment:
Can't you keep calling `Read(instream)` here? That might get rid of at least some of the test regressions.

Reply:
`Read()` does not return the value of an expression (which is the whole reason why this weird `_EVALRESULT` hack was there, I presume.

 I plan to replace the `READ_ALL_COMMANDS` by a `READ_ONE_COMMAND` from a stream soon (some work in progress is here: https://github.com/markuspf/gap/tree/inout-rewrite).

(Disclaimer: This is still part of libgap work, once the inout bit is done, we have incorporated every `libgap` bit and there'll be a PR to libgap in `master` proper)